go-errorlint suggest to exchange err.(validator.ValidationErrors) to errors.As

### Package version eg. v9, v10: 

	github.com/go-playground/locales v0.13.0
	github.com/go-playground/universal-translator v0.17.0
	github.com/go-playground/validator/v10 v10.2.0

### Issue, Question or Enhancement:

Golangci-lint (https://github.com/golangci/golangci-lint) comes with a errorlint linter (https://github.com/polyfloyd/go-errorlint). When I switched it on, I got this issue from the linter:

> validator/goplayground.go:35:21: type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors (errorlint)
> 		for _, e := range err.(validator.ValidationErrors) {

Is there a way to change err.(validator.ValidationErrors) on errors.As as suggested linter or it's a false-positive issue? Because I could not find examples other than err.(validator.ValidationErrors) in this repo.

### Code sample, to showcase or reproduce:

```go
func (v *Validator) Validate(toValidate interface{}) []string {
	err := v.validate.Struct(toValidate)
	var errors []string
	if err != nil {
		for _, e := range err.(validator.ValidationErrors) {
			errors = append(errors, e.Translate(v.translator))
		}
	}
	return errors
}
```

--- a/README.md
+++ b/README.md
@@ -51,7 +51,8 @@ Validator returns only InvalidValidationError for bad validation input, nil or V
 
 ```go
 err := validate.Struct(mystruct)
-validationErrors := err.(validator.ValidationErrors)
+var validationErrors validator.ValidationErrors
+errors.As(err, &validationErrors)
  ```
 
 Usage and documentation

--- a/_examples/translations/main.go
+++ b/_examples/translations/main.go
@@ -1,6 +1,7 @@
 package main
 
 import (
+	"errors"
 	"fmt"
 
 	"github.com/go-playground/locales/en"
@@ -70,13 +71,14 @@ func translateAll(trans ut.Translator) {
 	if err != nil {
 
 		// translate all error at once
-		errs := err.(validator.ValidationErrors)
-
-		// returns a map with key = namespace & value = translated error
-		// NOTICE: 2 errors are returned and you'll see something surprising
-		// translations are i18n aware!!!!
-		// eg. '10 characters' vs '1 character'
-		fmt.Println(errs.Translate(trans))
+		var errs validator.ValidationErrors
+		if errors.As(err, &errs) {
+			// returns a map with key = namespace & value = translated error
+			// NOTICE: 2 errors are returned and you'll see something surprising
+			// translations are i18n aware!!!!
+			// eg. '10 characters' vs '1 character'
+			fmt.Println(errs.Translate(trans))
+		}
 	}
 }
 
@@ -91,11 +93,12 @@ func translateIndividual(trans ut.Translator) {
 	err := validate.Struct(user)
 	if err != nil {
 
-		errs := err.(validator.ValidationErrors)
-
-		for _, e := range errs {
-			// can translate each error one at a time.
-			fmt.Println(e.Translate(trans))
+		var errs validator.ValidationErrors
+		if errors.As(err, &errs) {
+			for _, e := range errs {
+				// can translate each error one at a time.
+				fmt.Println(e.Translate(trans))
+			}
 		}
 	}
 }
@@ -119,11 +122,12 @@ func translateOverride(trans ut.Translator) {
 	err := validate.Struct(user)
 	if err != nil {
 
-		errs := err.(validator.ValidationErrors)
-
-		for _, e := range errs {
-			// can translate each error one at a time.
-			fmt.Println(e.Translate(trans))
+		var errs validator.ValidationErrors
+		if errors.As(err, &errs) {
+			for _, e := range errs {
+				// can translate each error one at a time.
+				fmt.Println(e.Translate(trans))
+			}
 		}
 	}
 }

